### DISCUSSION TITLE
Is it possible to listen for the search input events?

### DISCUSSION THREAD

**Discussion by davigiroux** [facebook/docusaurus#8765] [Q&A]
Hello!

So I want to have a way to track what is being searched in the search bar without interfering with the functionality itself. I basically need to trigger some events based on what is being typed. I tried swizzling the SearchBar component and the Search component from the NavBar, but did not manage to get what is being typed... Is there any way to accomplish that?

**Comment by slorber** [comment]
I'm assuming you use Algolia DocSearch, which relies on a component built by the Algolia team (`<DocSearchModal/>`).

The best is to get in touch with them directly to figure out if what you want to achieve and see how you can listen for typing events from their React props API?

cc @shortcuts

**Reply by davigiroux** [reply]
That makes sense. I'm actually using [easyops-cn/docusaurus-search-local](https://github.com/easyops-cn/docusaurus-search-local) so I'll reach out to them. Thank you!

**Reply by slorber** [reply]
Yes, it's the best. It's a third-party plugin so only they can give you the ability to listen for such events.
